palo_alto_cortex_xdr/help.md

```diff
@@ -0,0 +1,334 @@
+# Description
+
+Stop modern attacks with the industry’s first extended detection and response platform that spans your endpoint, network and cloud data.
+
+# Key Features
+
+* Get Endpoint Information
+* Isolate or Unisolate endpoints
+
+# Requirements
+
+* A Palo Alto Cortex XDR API key
+* A Palo Alto Cortex XDR API key ID
+* The URL to your Palo Alto Cortex XDR API instance
+
+The required connection information is available in the Cortex XDR web dashboard. Click the gear icon, click settings, and then click on API Keys on the left. 
+
+The API Key will be generated when you create a new API key. 
+
+The API Key ID is the value from the ID column
+
+To get the API URL, right click on your API key and pick generate examples. The generated example is a URL that should look similar to "https://api-yourorg.xdr.us.paloaltonetworks.com/api_keys/validate/". 
+In our example the API URL is "https://api-yourorg.xdr.us.paloaltonetworks.com" and should be copied.
+
+
+# Documentation
+
+## Setup
+
+The connection configuration accepts the following parameters:
+
+|Name|Type|Default|Required|Description|Enum|Example|
+|----|----|-------|--------|-----------|----|-------|
+|api_key|credential_secret_key|None|True|The Cortex XDR API Key that is generated when creating a new key|None|1234123412341234asdfasdfasdfasdfasdf1234123412341234123412341234asdfasdfasdfasdfasdf123412341234123412341234asdfasdfasdfasdfasdf|
+|api_key_id|int|None|True|The API Key ID shown in the Cortex XDR API Keys table in settings. e.g. 1, 2, 3|None|1|
+|security_level|string|Standard|True|The Security Level of the key provided. This can be found in the API Key settings table in the Cortex XDR settings|['Advanced', 'Standard']|Standard|
+|url|string|None|True|Cortex XDR API URL|None|https://api-example.xdr.us.paloaltonetworks.com/|
+
+Example input:
+
+```
+{
+  "api_key": "1234123412341234asdfasdfasdfasdfasdf1234123412341234123412341234asdfasdfasdfasdfasdf123412341234123412341234asdfasdfasdfasdfasdf",
+  "api_key_id": 1,
+  "security_level": "Standard",
+  "url": "https://api-example.xdr.us.paloaltonetworks.com/"
+}
+```
+
+## Technical Details
+
+### Actions
+
+#### Block File
+
+This action is used to add a file to the block list.
+
+##### Input
+
+|Name|Type|Default|Required|Description|Enum|Example|
+|----|----|-------|--------|-----------|----|-------|
+|comment|string|File blocked by InsightConnect|True|String that represents additional information regarding the action|None|File blocked by InsightConnect|
+|file_hash|string|None|True|A SHA256 file hash|None|275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f|
+|incident_id|string|None|False|If this is related to an incident, the ID should be entered here|None|5|
+
+Example input:
+
+```
+{
+  "comment": "File blocked by InsightConnect",
+  "file_hash": "275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f",
+  "incident_id": 5
```

Comment:
Does incident_id value need to be enclosed with quotes since its type is string or it does not matter?f

Reply:
That's a good catch.